### PR TITLE
Add OpenAPI diff action

### PR DESCRIPTION
**Pull request by avoiney-ledger**
Example usage:

```yaml
name: Generate OpenApi Diff

on:
  pull_request:
  workflow_dispatch:
    inputs:
      base_ref:
        description: "Base branch for the diff"
        required: true
        default: "main"
      head_ref:
        description: "Head branch for the diff"
        required: false

jobs:
  generate-diff:
    name: Generate OpenAPI diff
    runs-on: ubuntu-latest

    steps:
      - uses: LedgerHQ/actions/python-app/openapi-diff@main
        with:
          base_ref: ${{github.event.inputs.base_ref || github.base_ref}}
          head_ref: ${{github.event.inputs.head_ref || github.head_ref}}
          doc_output: doc/api-spec.yaml
        env:
          PYPI_DEPLOY_TOKEN: "${{ secrets.PYPI_DEPLOY_TOKEN }}"

### CODE REVIEW

File: python-app/api-diff/action.yaml [hunk 1]
```diff
@@ -0,0 +1,98 @@
+name: OpenApi Diff
+description: Generate and comment OpenAPI Diff
+
+inputs:
+  base_ref:
+    description: "Base branch for the diff"
+    required: true
+    default: "main"
```
Comment (noudin-ledger) [inline python-app/api-diff/action.yaml]: Is base ref required as there's a default value?
We want not to have it mandatory if we can always diff with main ?

File: python-app/openapi-diff/action.yaml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+
+runs:
+  using: "composite"
+
+  steps:
+    - name: set up python
+      uses: actions/setup-python@v4
+      with:
+        python-version: "3.11"
```
Comment (noudin-ledger) [inline python-app/openapi-diff/action.yaml:21]: Maybe you can add a cache key here so we gain time by not re downloading all libraries over and over again.
Documentation says:
```
- uses: actions/setup-python@v4
  with:
    python-version: '3.9'
    cache: 'pipenv' # caching pipenv dependencies
```

File: python-app/openapi-diff/action.yaml [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+    - name: install dependencies
+      run: pipenv sync
+      shell: bash
+
+    - name: Generate OpenAPI for base
+      run: |
+        pipenv run openapi
+        mkdir specs
+        cp doc/api-spec.yaml ./specs/base-spec.yaml
```
Comment (noudin-ledger) [inline python-app/openapi-diff/action.yaml:45]: Maybe this path should be an argument so it adapts to different repos with different paths and names

File: python-app/openapi-diff/action.yaml [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+      run: |
+        npm install -g bump-cli
+      shell: bash
+
+    - name: create outputs directory
+      run: |
+        mkdir outputs
+      shell: bash
+
+    - name: generate mardown diff
```
Comment (noudin-ledger) [inline python-app/openapi-diff/action.yaml:76]: markdown*

Comment (avoiney-ledger) [reply python-app/openapi-diff/action.yaml:76]: indeed :pray:

File: python-app/openapi-diff/action.yaml (hunk 2 above)
Comment (noudin-ledger) [reply python-app/openapi-diff/action.yaml:24]: Yes indeed, agreed

**Review by noudin-ledger — COMMENTED** [review]
Can you just test it to ensure it works?
Or did you do it already?

**Comment by avoiney-ledger** [comment]
> Can you just test it to ensure it works? Or did you do it already?

Already tested:
- pipenv repo: https://github.com/LedgerHQ/vault-lam/pull/385
- pdm repo: https://github.com/LedgerHQ/vault-authentication-service/pull/31

**Review by noudin-ledger — APPROVED** [review]

**Pull request merged**